client/src/components/Profile/ProfileTabSelector/index.tsx

```diff
@@ -14,18 +16,27 @@ interface ProfileTabSelectorProps {
     setActiveTab: (tab: string) => void
 }
 
-export const ProfileTabSelector = (props: ProfileTabSelectorProps) => {
+export const ProfileTabSelector = observer((props: ProfileTabSelectorProps) => {
     const { activeTab, setActiveTab } = props
+    const navigate = useNavigate()
 
-    const tabSwitcher = (tab: string): void => {
+    const tabSwitcher = async (tab: string): Promise<void> => {
         setActiveTab(tab)
-        if (tab === tabs.logOut) {
-            setActiveTab(tabs.logOut)
-            userStore.setUser(null)
-            userStore.setIsAuth(false)
+    }
+
+    const logOut = async () => {
+        if (activeTab === tabs.logOut) {
+            await localStorage.removeItem('token')
+            await userStore.setUser(null)
+            await userStore.setIsAuth(false)
+            navigate('/')
         }
     }
 
+    useEffect(() => {
+        logOut()
```

Comment:
maybe 
if (activeTab === tabs.logOut) { 
 logOut()
}
and remove if from logout function
?